### PR TITLE
fix(curriculum): improved test for linked list challenge

### PR DESCRIPTION
**Pull request by MattDClarke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

The test that checks that `head` is reassigned to the new node, when the given index is 0, does not check that the newly added elements next is pointing to the node that was previously the `head`. This pull request adds this test.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/data-structures/add-elements-at-a-specific-index-in-a-linked-list.md [hunk 1]
```diff
@@ -24,8 +24,8 @@ assert(
     var test = new LinkedList();
     test.add('cat');
     test.add('dog');
-    test.addAt(0, 'cat');
-    return test.head().element === 'cat';
+    test.addAt(0, 'fish');
+    return (test.head().element === 'fish') && (test.head().next === 'cat');
```
Comment (gikf) [inline curriculum/challenges/english/10-coding-interview-prep/data-structures/add-elements-at-a-specific-index-in-a-linked-list.md:28]:
```suggestion
    return test.head().element === 'fish' && test.head().next.element === 'cat';
```
Checking the `.element` of the `next` should fix issue with failing test.

**Review by gikf — DISMISSED** [review]

**Review by ejcheng — APPROVED** [review]

**Review by gikf — APPROVED** [review]
LGTM :+1:

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Comment by ShaunSHamilton** [comment]
Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**